addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -147,11 +147,24 @@ async function _buildImgElement(src) {
     if (!(src in _buildImgElementCache)) {
         _buildImgElementCache[src] = (async () => {
             if (src.split('.').pop() === 'svg') {
-                const response = await window.fetch(src);
-                const text = await response.text();
-                const parser = new window.DOMParser();
-                const xmlDoc = parser.parseFromString(text, 'text/xml');
-                return xmlDoc.getElementsByTagName('svg')[0];
+                try {
+                    const response = await window.fetch(src);
+                    const text = await response.text();
+                    const parser = new window.DOMParser();
+                    const xmlDoc = parser.parseFromString(text, 'text/xml');
+                    return xmlDoc.getElementsByTagName('svg')[0];
+                } catch (e) {
+                    // In some tours, the tour finishes before the fetch is done
+                    // and when a tour is finished, the python side will ask the
+                    // browser to stop loading resources. This causes the fetch
+                    // to fail and throw an error which crashes the test even
+                    // though it completed successfully.
+                    // So return an empty SVG to ensure everything completes
+                    // correctly.
+                    if (e.message?.startsWith("Failed to fetch")) {
```

Comment:
- You probably should not try..catch the whole block
- Checking "Failed to fetch" explicitly probably only works on one browser, for now